packages/frontend/src/components/DataViz/config/CartesianChartFieldConfiguration.tsx

```diff
@@ -120,32 +121,67 @@ const XFieldAxisConfig = ({
 
     return (
-        <FieldReferenceSelect
-            clearable
-            data={xLayoutOptions.map((x) => ({
-                value: x.reference,
-                label: x.reference,
-            }))}
-            value={field?.reference ?? null}
-            placeholder="Select X axis"
-            onChange={(value) => {
-                if (!value) {
-                    dispatch(actions.removeXAxisField());
-                } else dispatch(actions.setXAxisReference(value));
-            }}
-            error={
-                field?.reference &&
-                xLayoutOptions.find((x) => x.reference === field.reference) ===
-                    undefined &&
-                `Column "${field.reference}" does not exist. Choose another`
-            }
-            fieldType={
-                (field?.reference &&
-                    columns?.find((x) => x.reference === field.reference)
-                        ?.type) ||
-                DimensionType.STRING
-            }
-        />
+        <>
+            <FieldReferenceSelect
+                clearable
+                data={xLayoutOptions.map((x) => ({
+                    value: x.reference,
+                    label: x.reference,
+                }))}
+                value={field?.reference ?? null}
+                placeholder="Select X axis"
+                onChange={(value) => {
+                    if (!value) {
+                        dispatch(actions.removeXAxisField());
+                    } else dispatch(actions.setXAxisReference(value));
+                }}
+                error={
+                    field?.reference &&
+                    xLayoutOptions.find(
+                        (x) => x.reference === field.reference,
+                    ) === undefined &&
+                    `Column "${field.reference}" does not exist. Choose another`
+                }
+                fieldType={
+                    (field?.reference &&
+                        columns?.find((x) => x.reference === field.reference)
+                            ?.type) ||
+                    DimensionType.STRING
+                }
+            />
+            {field?.reference && (
+                <Config.Group>
+                    <Config.Label>Sort by</Config.Label>
+                    <Select
+                        radius="md"
+                        placeholder="Select sort option"
+                        data={[
+                            {
+                                value: SortByDirection.ASC,
+                                label: 'Ascending',
+                            },
+                            {
+                                value: SortByDirection.DESC,
+                                label: 'Descending',
+                            },
+                        ]}
+                        onChange={(value) => {
+                            if (value) {
+                                const direction = value as SortByDirection;
```

Comment:
@joaoviana how do we type mantine form fields?

Reply:
I believe this will work:
```ts
onChange={(value: SortByDirection) => {
```

because `data` is static